app/controllers/api/public/v1/courses_controller.rb

```diff
@@ -7,24 +7,40 @@ class CoursesController < API::Public::V1::ApplicationController
         def index
           render jsonapi: paginate(courses),
                  include: include_param,
-                 meta: { count: courses.count('course.id') },
+                 meta: { count: cached_course_count },
                  class: API::Public::V1::SerializerService.call
         rescue ActiveRecord::StatementInvalid
           render json: { status: 400, message: 'Invalid changed_since value, the format should be an ISO8601 UTC timestamp, for example: `2019-01-01T12:01:00Z`' }.to_json, status: :bad_request
         end
 
         private
 
+        def cached_course_count
+          year = permitted[:recruitment_cycle_year] || RecruitmentCycle.current.year
+
+          if permitted[:no_cache]
+            courses.count('course.id')
+          else
+            Rails.cache.fetch("api_course_count_#{year}", expires_in: 5.minutes) do
```

Comment:
We can let Rails handle the stringification our cache key here for our convenience. Not a blocker.

```suggestion
            Rails.cache.fetch([:api_course_count, year], expires_in: 5.minutes) do
```